stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/userreactions/UserReactions.kt

```diff
@@ -0,0 +1,111 @@
+package io.getstream.chat.android.compose.ui.components.userreactions
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.GridCells
+import androidx.compose.foundation.lazy.LazyVerticalGrid
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import io.getstream.chat.android.compose.R
+import io.getstream.chat.android.compose.previewdata.PreviewUserReactionData
+import io.getstream.chat.android.compose.state.userreactions.UserReactionItemState
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+
+/**
+ * Represent a section with a list of reactions left for the message.
+ *
+ * @param items The list of user reactions to display.
+ * @param modifier Modifier for styling.
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+public fun UserReactions(
+    items: List<UserReactionItemState>,
+    modifier: Modifier = Modifier,
+    itemContent: @Composable (UserReactionItemState) -> Unit = {
+        UserReactionItem(
+            item = it,
+            modifier = Modifier,
+        )
+    },
+) {
+    val reactionCount = items.size
+
+    val reactionCountText = LocalContext.current.resources.getQuantityString(
+        R.plurals.stream_compose_message_reactions,
+        reactionCount,
+        reactionCount
+    )
+
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = modifier.background(ChatTheme.colors.barsBackground)
+    ) {
+        Text(
+            text = reactionCountText,
+            style = ChatTheme.typography.title3Bold,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis,
+            color = ChatTheme.colors.textHighEmphasis
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        if (reactionCount > 0) {
+            BoxWithConstraints(
```

Comment:
Unfortunately, had to calculate the number of column here manually.

Reply:
Would using let's say `GridCells.Adaptive(72.dp)` achieve the wanted result?

Reply:
Would using let's say `GridCells.Adaptive(ChatTheme.dimens.userReactionItemWidth)` achieve the wanted result without manual calculation?

Reply:
Yes, but the cell won't be centered for 1-3 items. Maybe you have some working code snippet? Basically the requirements are:
- Behave like flow layout when N > Nmax (first row fully filled, the next row starts from the left)
- Center items when N < Nmax (1 items - centered, 2 items - centered, etc.)

Reply:
Would a `wrapContent` and an `align(Center)` in combination of the grid work?

Reply:
Doesn't work, seems like a Compose bug